packages/test-server/src/types.ts

```diff
@@ -0,0 +1,189 @@
+export interface EnvelopeHeaders {
+  event_id: string;
+  dsn: string;
+  client: string;
+  version: number;
+  remote_addr: string;
+  forwarded_for: string;
+  user_agent: string;
+  sent_at: string;
+}
+
+export interface ItemHeaders {
+  type: string;
+  length: number;
+  content_type?: string;
+  attachment_type?: string;
+  filename?: string;
+}
+
+export interface Item {
+  headers: ItemHeaders;
+  payload?: Event;
+}
+
+export interface Envelope {
+  headers: EnvelopeHeaders;
+  items: Item[];
+}
+
+export interface OkEvent {
+  Ok: Envelope;
+}
+
+export interface ErrEvent {
+  Err: any;
+}
+
+export type ServerEvent = OkEvent | ErrEvent;
+
+export interface Event {
```

Comment:
Tbh these types should just be in `@sentry/types` but we can deal with that afterwards.

Reply:
I auto generated most of these types from all the data submitted through the test server using https://quicktype.io and made some minor fixes.

They differ slightly from `@sentry/types` in places because Relay does some normalisation. The only example that springs to mind is `tags` becoming `string[][]` after Relay rather than `Record<string, string>`.

Reply:
Ah yes you are right about the small typing mismatches. I explored the JSON schema relay exported a while back https://github.com/getsentry/sentry-javascript/pull/2802 and hit those.

Reply:
We can codemod those small differences out 😄